File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        ios_base::sync_with_stdio(false);
        int ans=nums.size();
        sort(nums.begin(),nums.end());
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=i){
                ans=i;
                break;
            }
        }
        return ans;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int xor = 0;
        int n = nums.length;

        // XOR all indices and values in the array
        for (int i = 0; i < n; i++) {
            xor ^= i ^ nums[i];
        }

        // XOR the last index n
        xor ^= n;

        return xor;
    }
}
